theano/gof/cc.py

```diff
@@ -476,11 +476,12 @@ def apply_policy(policy, r, name, sub):
         C{policy[0](r) + policy[1](r) + ...}.
 
     """
-    if isinstance(policy, (list, tuple)):
-        ret = ""
-        for sub_policy in policy:
-            ret += sub_policy(r, name, sub)
-        return ret
+    if not isinstance(policy, (list, tuple)):
+        policy = [policy]
+    ret = ""
+    for sub_policy in policy:
+        ret += sub_policy(r, name, sub)
+    return ret
     return policy(r, name, sub)
```

Comment:
The changed code is functionally equivalent to the old code, so it won't break anything.  However having two return statements one after the other is superfluous.

You can either get rid of the second one or revert the code I guess.